src/createRestyleFunction.ts

```diff
@@ -55,11 +55,37 @@ const createRestyleFunction = <
         typeof themeKey === 'string' &&
         typeof property === 'string'
       ) {
+        /*
+        When using variants, each variant might have similar styles with breakpoints. For example, below we have variants with different padding that each have breakpoints:
+          spacingVariant: {
+            defaults: {},
+            noPadding: {
+              phone: 'none',
+              tablet: 'none',
+            },
+            mediumPadding: {
+              phone: 'm',
+              tablet: 'm',
+            }
+          }
+
+          When a variant is an object, the `property` that is passed into this code block is also an object. Since we pass this object as a string to the getMemoizedMapHashKey(),
+          the hash key ends up looking something like this: 1334x750-spacing-padding-[object Object].
+
+          Since it ends with [object Object], it doesn't actually know what that object is. Even though it can be {phone: 'none', tablet: 'none'} or {phone: 'm', tablet: 'm'}, the memoized value just returns [object Object].
+          This causes issues because we then try to get the second object's value, it actually just returns the first object's value there is already a memoized  hash key with [object Object].
+          To solve this, we stringify the variants such that their hash key looks like this: 1334x750-spacing-padding-{"phone":"m","tablet":"m"}.
+          Now, each variant will have a unique hash key so we won't accidentally be grabbing an already memoized hash key that returns an incorrect value.
+        */
+        const value =
+          typeof props[property] === 'object'
+            ? JSON.stringify(props[property])
```

Comment:
I worry about `JSON.stringify` speed. This object should always have breakpoints, right? Can't we instead loop through breakpoints and constructing the string ourselves? In pseudocode:
```
let propertyString = ""
for breakpoint in theme.breakpoints {
  propertyString += props[property][breakpoint]
}
```

Reply:
> If we'd like a delimiter between the object's properties, another option would be to do the following:

I think it's ok without the delimiter, the key for the hash map is an implementation detail and not something users of the library should be aware of.